File: backend/src/main/java/com/gxs/bank/repository/LoanRepository.java
package com.gxs.bank.repository;

import com.gxs.bank.model.Loan;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import java.util.List;
import java.util.UUID;

@Repository
public interface LoanRepository extends JpaRepository<Loan, UUID> {
    List<Loan> findByUserId(UUID userId);
    List<Loan> findByStatus(Loan.Status status);
    List<Loan> findByUserIdAndStatus(UUID userId, Loan.Status status);
    long countByStatus(Loan.Status status);
}

File: backend-python/com/gxs/bank/repository/LoanRepository.py
from sqlalchemy import func

from com.gxs.bank.model.Loan import Loan
from com.gxs.bank.repository._base import BaseRepository


class LoanRepository(BaseRepository):
    def __init__(self, db):
        super().__init__(db, Loan)

    def findByUserId(self, userId: str):
        return self.db.query(Loan).filter(Loan.userId == userId).all()

    def findByStatus(self, status):
        return self.db.query(Loan).filter(Loan.status == status).all()

    def findByUserIdAndStatus(self, userId: str, status):
        return self.db.query(Loan).filter(Loan.userId == userId, Loan.status == status).all()

    def countByStatus(self, status):
        return int(self.db.query(func.count(Loan.id)).filter(Loan.status == status).scalar() or 0)
